File: ShiftSync.API/Models/Entities/AttendanceBreak.cs
using ShiftSync.API.Models.Enums;

namespace ShiftSync.API.Models.Entities;

public class AttendanceBreak
{
    public int Id { get; set; }

    public int AttendanceId { get; set; }
    public Attendance Attendance { get; set; } = null!;

    public int BreakTypeId { get; set; }
    public BreakType BreakType { get; set; } = null!;

    public DateTime RequestTime { get; set; } = DateTime.UtcNow;
    public DateTime? StartTime { get; set; }
    public DateTime? EndTime { get; set; }

    public BreakStatus Status { get; set; } = BreakStatus.Approved;

    public string? Note { get; set; }
}

File: ShiftSync.Web/src/types/shiftsync/attendanceBreak.ts
export enum BreakStatus {
  Approved = 1,
  WaitingQueue = 2,
  Rejected = 3,
  Completed = 4,
}

export interface AttendanceBreak {
  id: number;
  attendanceId: number;
  breakTypeId: number;
  breakTypeName?: string;
  requestTime: string;
  startTime?: string;
  endTime?: string;
  status: BreakStatus;
  note?: string;
  userFullName?: string;
}

export interface UpdateBreakStatusDto {
  status: BreakStatus;
  note?: string;
}

export interface AttendanceBreakFilter {
  pageNumber?: number;
  pageSize?: number;
  attendanceId?: number;
  status?: BreakStatus;
  breakTypeId?: number;
}
